craft_archives/repo/package_repository.py

```diff
@@ -22,21 +22,24 @@
 from typing import Any, Dict, List, Mapping, Optional, Union
 from urllib.parse import urlparse
 
-import pydantic
 from overrides import overrides  # pyright: ignore[reportUnknownVariableType]
 from pydantic import (
     AnyUrl,
-    ConstrainedStr,
+    BaseModel,
+    ConfigDict,
+    Field,  # pyright: ignore[reportUnknownVariableType]
```

Comment:
we should check whether these annotations are still necessary

Reply:
We can probably make it more idiomatic, but since these work for now I'm bumping that to later. I've made this issue for it: https://github.com/canonical/craft-archives/issues/118